### PR TITLE
schedule: use config interface for scheduler controller

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #5839.

### What is changed and how does it work?

It needs #6773 to merge first.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by nolouch — APPROVED** [review]

File: server/config/persist_options.go [hunk 1]
```diff
@@ -701,6 +712,22 @@ func (o *PersistOptions) AddSchedulerCfg(tp string, args []string) {
...
+	v := o.GetScheduleConfig().Clone()
+	for i, schedulerCfg := range v.Schedulers {
+		if tp == schedulerCfg.Type {
+			if IsDefaultScheduler(tp) {
+				schedulerCfg.Disable = true
+				v.Schedulers[i] = schedulerCfg
+			} else {
+				v.Schedulers = append(v.Schedulers[:i], v.Schedulers[i+1:]...)
+			}
+			o.SetScheduleConfig(v)
```
Comment (HuSharp) [inline server/config/persist_options.go:726]: Do we need to return directly here?

Comment (rleungx) [reply server/config/persist_options.go:726]: nice catch

**Review by HuSharp — APPROVED** [review]
LGTM!

**Pull request merged**